packages/react/slider/src/Slider.tsx

```diff
@@ -390,70 +347,24 @@ type SliderPartPrimitive = Polymorphic.ForwardRefComponent<
   SliderPartOwnProps
 >;
 
-const SliderPart = React.forwardRef((props, forwardedRef) => {
+const SliderImpl = React.forwardRef((props, forwardedRef) => {
   const {
     as = SLIDER_DEFAULT_TAG,
-    onSlideMouseDown,
-    onSlideMouseMove,
-    onSlideMouseUp,
-    onSlideTouchStart,
-    onSlideTouchMove,
-    onSlideTouchEnd,
+    onSlideStart,
+    onSlideMove,
+    onSlideEnd,
     onHomeKeyDown,
     onEndKeyDown,
     onStepKeyDown,
     ...sliderProps
   } = props;
-
-  const handleSlideMouseMove = useCallbackRef(onSlideMouseMove);
-  const handleSlideTouchMove = useCallbackRef(onSlideTouchMove);
-  const removeMouseEventListeners = useCallbackRef(() => {
-    document.removeEventListener('mousemove', handleSlideMouseMove);
-    document.removeEventListener('mouseup', removeMouseEventListeners);
-    onSlideMouseUp();
-  });
-  const removeTouchEventListeners = useCallbackRef(() => {
-    document.removeEventListener('touchmove', handleSlideTouchMove);
-    document.removeEventListener('touchend', removeTouchEventListeners);
-    onSlideTouchEnd();
-  });
```

Comment:
So just so I'm clear, we don't need the document handlers because we now use pointer events with pointer capture? So even if we move quickly and our cursor escapes the thumb it will still work?

Reply:
We were using document events before so that they were only bound when a drag started. With pointer events we can check `hasPointerCapture` to determine if they're dragging instead.

Reply:
nice 👍